sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -1324,15 +1329,15 @@ case class Concat(children: Seq[Expression]) extends Expression {
   since = "2.4.0")
 case class Flatten(child: Expression) extends UnaryExpression {
 
-  private val MAX_ARRAY_LENGTH = ByteArrayMethods.MAX_ROUNDED_ARRAY_LENGTH
-
+  @transient
   private lazy val childDataType: ArrayType = child.dataType.asInstanceOf[ArrayType]
```

Comment:
we should only use `lazy val` if it can save some heavy computation. For things like this, I think `def` is good enough